app/controllers/concerns/date_range_parser.rb

```diff
@@ -0,0 +1,48 @@
+require 'active_support/concern'
+
+module Concerns::DateRangeParser
+  extend ActiveSupport::Concern
+
+  included do
+    def parse_date_range(date_range)
+      date_to_range(date_range)
```

Comment:
This method returns a hash with two attributes. A `DateRange` seems like a very useful class for our Domain. What extracting a class `/model/date_range`?

Reply:
Extracting a class like DateRange would have the following benefits in the short term:

1. The concern does not return a Hash that you later on have to inspect to extract `to` and `from`
2. We will have to implement shortly retrieving the metrics for the month before, so we can always add a method to perform this operation

We would be able to move from:

```ruby
    date_range = parse_date_range(params[:date_range])
    from = date_range[:from]
    to = date_range[:to]

    service_params = {
      base_path: params[:base_path],
      from: from,
      to: to,
      metrics: DEFAULT_METRICS
    }
```

to 

```ruby
  service_params = {
   base_path: params[:base_path],
   date_range: parse_date_range(params),
   metrics: DEFAULT_METRICS
  } 
```